RecoEgamma/EgammaElectronAlgos/src/ElectronSeedGenerator.cc

```diff
@@ -163,29 +164,10 @@ ElectronSeedGenerator::ElectronSeedGenerator(const edm::ParameterSet &pset, cons
                useRecoVertex_) {}
 
 void ElectronSeedGenerator::setupES(const edm::EventSetup &setup) {
-  // get records if necessary (called once per event)
-  bool tochange = false;
-
-  if (cacheIDMagField_ != setup.get<IdealMagneticFieldRecord>().cacheIdentifier()) {
-    setup.get<IdealMagneticFieldRecord>().get(magField_);
-    cacheIDMagField_ = setup.get<IdealMagneticFieldRecord>().cacheIdentifier();
-    propagator_ = std::make_unique<PropagatorWithMaterial>(alongMomentum, .000511, &(*magField_));
-    tochange = true;
-  }
-
-  if (cacheIDTrkGeom_ != setup.get<TrackerDigiGeometryRecord>().cacheIdentifier()) {
-    cacheIDTrkGeom_ = setup.get<TrackerDigiGeometryRecord>().cacheIdentifier();
-    setup.get<TrackerDigiGeometryRecord>().get(trackerGeometry_);
-    tochange = true;  //FIXME
-  }
-
-  if (tochange) {
-    matcher_.setES(magField_.product(), trackerGeometry_.product());
-  }
+  matcher_.setES(setup.getData(magFieldToken_), setup.getData(trackerGeometryToken_));
```

Comment:
PixelHitMatcher::setES creates two propagators, which seems rather costly to just start doing it every event.
I suggest to revert

Reply:
Okay, I don't think that constructing the propagator for every event is costy (it is already done at the `ElectronNHitsSeedProducer` for the HLT), but it's a fair point. I won't fully revert because we still want to use the ESGetTokens I guess, but I can bring back the `cacheIdentifier()` check.

Reply:
I use now the `[edm::ESWatcher](https://github.com/cms-sw/cmssw/blob/7ac438051e7ba3c56de9894520931b34320ce730/FWCore/Framework/interface/ESWatcher.h)`, which does the same thing as before (checking the cache ID).

I don't think the ESWatcher itself is migrated to esConsumes yet, though. Will this be done in the future or should we avoid using the `ESWatcher`?